api/src/shared/validations/isState.ts

```diff
@@ -0,0 +1,23 @@
+import { STATES, STATE_CODES } from '../constants/states'
+import { z } from 'zod'
+
+/**
+ * Validaiton function intended to be used in a zod schema
+ * @example
+ * const schema = z.object({
+ *   name: z.string().optional(),
+ *   state: isState().optional() /// etc
+ * })
+ */
+export function isState(stateLength: 'long' | 'short' = 'short') {
+  return z.string().refine(
+    (value: any) => {
+      const input = String(value).toLowerCase()
```

Comment:
Should this be `any`?

Shouldn't whatever this is using be sending in a presumed `string` parameter, right?  I don't think we need runtime type-casting when TS _should_ tell us if we're linking up something from the Prisma model that we've defined as a different type, no? 🤔